src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,431 @@
+After deploying the resources, you can test the data ingestion process by adding an item to the `Todo` table. However, before doing that, let's verify that the pipeline has been set up correctly.
+
+In the AWS console, navigate to OpenSearch and then to the pipelines section. You should find your configured pipeline and review its settings to ensure they match your expectations:
+
+![A screenshot of OpenSearch OSIS pipeline](/images/gen2/opensearch-integration/OpenSearch_pipeline.png)
+
+You can also check this in the DynamoDB console by going to the Integrations section of the tables.
+
+![A screenshot of DynamoDB integration](/images/gen2/opensearch-integration/OpenSearch_DynamoDB_integration.png)
+
+## Step 4: Expose new queries on OpenSearch
+
+### Step 4a:Add OpenSearch Datasource to backend
+
+First, Add the OpenSearch data source to the data backend.
+
+``` ts title="amplify/backend.ts"
+
+const osDataSource = backend.data.addOpenSearchDataSource(
+  "osDataSource",
+  openSearchDomain
+);
+```
+### Step 4b: Create Resolver and attach to query
+
+Let's create the search resolver. Create a new file named `amplify/data/searchBlogResolver.ts` and paste the following code. For additional details please refer to [Amazon OpenSearch Service Resolvers](https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-elasticsearch-resolvers-js.html)
+
+```ts title="amplify/data/searchBlogResolver.ts"
+
+import { util } from "@aws-appsync/utils";
+
+/**
+ * Searches for documents by using an input term
+ * @param {import('@aws-appsync/utils').Context} ctx the context
+ * @returns {*} the request
+ */
+
+export function request(ctx) {
+  return {
+    operation: "GET",
+    path: "/todo/_search",
+  };
+}
+
+/**
+ * Returns the fetched items
+ * @param {import('@aws-appsync/utils').Context} ctx the context
+ * @returns {*} the result
+ */
+
+export function response(ctx) {
+  if (ctx.error) {
+    util.error(ctx.error.message, ctx.error.type);
+  }
+  return ctx.result.hits.hits.map((hit) => hit._source);
+}
+
+
+```
+
+### Step 4c: Add the AppSync Resolver for the Search Query
+
+Update the schema and add a searchTodo query.
+
+``` ts title="amplify/data/resource.ts"
+
+const schema = a.schema({
+  Todo: a
+    .model({
+      content: a.string(),
+      done: a.boolean(),
+      priority: a.enum(["low", "medium", "high"]),
+    })
+    .authorization([a.allow.public()]),
+
+  //highlight-start  
+    searchTodos: a
+    .query()
+    .returns(a.ref("Todo").array())
+    .authorization([a.allow.public()])
+    .handler(
+      a.handler.custom({
+        entry: "./searchBlogResolver.js",
+        dataSource: "osDataSource",
+      })
+    ),
+  //highlight-end
+
+});
+```
+
+Once you've deployed the resources, you can check the AppSync console to confirm the changes. Run the searchTodo query and review the returned results to verify their accuracy.
+
+![A screenshot of AppSync console](/images/gen2/opensearch-integration/opensearch_appsync_console.png)
```

Comment:
Could we get a more descriptive alt text for this? Maybe something like,

`A screenshot of the AppSync console, showing a query for .... with the results of ... on the right`

Something like that, maybe?

Reply:
updated alt text.